app/validators/event_name_validator.rb

```diff
@@ -0,0 +1,11 @@
+##
+# Validates that the Event.name matches the class of the Event.handleable.
+# For example if Event.handleable is a Events::PullRequest object the Event.name
+# is expected to be 'pull_request'
+class EventNameValidator < ActiveModel::Validator
+  def validate(event)
+    return if event.name == event.handleable_type_name
+
+    event.errors[:event_name_validator] << 'Event.name must match Event.handleable type'
```

Comment:
Maybe the error should be attached to the `name` attribute of the event instead of `event_name_validator`